core/errors/wrappedErrors_test.go

```diff
@@ -0,0 +1,107 @@
+package errors
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestWrapError(t *testing.T) {
+	t.Parallel()
+
+	we := WrapError(nil)
+	require.NotNil(t, we)
+	assert.Equal(t, nil, we.GetLastError())
+	assert.True(t, strings.Contains(we.Error(), "/core/errors/wrappedErrors_test.go"))
+	assert.True(t, strings.Contains(we.Error(), nilPlaceholder))
+
+	definedError := errors.New("defined errors")
+	we = WrapError(definedError)
+	require.NotNil(t, we)
+	assert.Equal(t, definedError, we.GetLastError())
+	assert.True(t, strings.Contains(we.Error(), "/core/errors/wrappedErrors_test.go"))
+	assert.True(t, strings.Contains(we.Error(), definedError.Error()))
+	assert.True(t, we.Is(definedError))
+
+	we1 := WrapError(definedError)
+	we2 := WrapError(we1)
+	require.NotNil(t, we2)
+	assert.Equal(t, definedError, we2.GetLastError())
+	assert.True(t, strings.Contains(we2.Error(), "/core/errors/wrappedErrors_test.go"))
+	assert.True(t, strings.Contains(we2.Error(), we1.Error()))
+	assert.True(t, we2.Is(definedError))
+}
+
+func TestWrappableError_WrapWithMessage(t *testing.T) {
+	t.Parallel()
+
+	we := WrapError(nil)
+	we = we.WrapWithMessage("")
+	require.NotNil(t, we)
+	assert.NotNil(t, we.GetLastError())
+	assert.True(t, strings.Contains(we.Error(), "/core/errors/wrappedErrors_test.go"))
+	assert.True(t, strings.Contains(we.Error(), nilPlaceholder))
+
+	we = WrapError(nil)
+	msg := "a message"
+	we = we.WrapWithMessage(msg)
+	require.NotNil(t, we)
+	assert.NotNil(t, we.GetLastError())
+	assert.True(t, strings.Contains(we.Error(), "/core/errors/wrappedErrors_test.go"))
```

Comment:
you could have assigned the `we.Error()` to a variable, and then use it for checks

Reply:
done and then reverted as it defeats the purpose of easily understanding the tests.